dids/diddht/bencode/bencode.go

```diff
@@ -0,0 +1,260 @@
+package bencode
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+)
+
+const (
+	DictionaryPrefix = 'd'
+	IntegerPrefix    = 'i'
+	ListPrefix       = 'l'
+	EndSuffix        = 'e'
+)
+
+// Marshal encodes the given input into a Bencode formatted byte array.
+// More information about Bencode can be found at:
+// https://wiki.theory.org/BitTorrentSpecification#Bencoding
+func Marshal(input any) ([]byte, error) {
+	switch v := input.(type) {
+	case string:
+		encoded := fmt.Sprintf("%d:%s", len(v), v)
+
+		return []byte(encoded), nil
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
+		encoded := fmt.Sprintf("%s%d%s", string(IntegerPrefix), v, string(EndSuffix))
```

Comment:
You can use `%c` for characters if you prefer.

Reply:
ah yeah much cleaner! done: https://github.com/TBD54566975/web5-go/commit/b5bec440d2da58e8825823f2d315678fbb09368d